### ISSUE TITLE
We have lost the previous tasks in the WebServer

### ISSUE THREAD

**Issue by PhoenixChamb** [apache/airflow#31179]
### Apache Airflow version

2.6.0

### What happened

After upgrading Airflow to 2.6.0 from 2.5.1

We are losing the previous tasks on the Airflow web server GRID . These tasks are executed correctly however they are not displayed.
![image](https://github.com/apache/airflow/assets/122443108/917eae4a-5f06-48e4-a632-2673ccab635d)

The behavior is curious, since the previous tasks are finished. An example is that if I mark the last one as Failed. The Grid is updated and shows me the tasks again.

![image](https://github.com/apache/airflow/assets/122443108/4e4233f0-a938-44d1-aec6-00d1af9ba3c8)

Also if I select this option. You can see that the tasks are found, but they are never shown.

![image](https://github.com/apache/airflow/assets/122443108/b299b9ac-ff00-4d86-9647-d82f5c466940)

But finally I do not do a clear or anything, mysteriously the tasks in the Grid are loaded correctly (sometimes are loaded with the correct state "sucessfull" other time as in the image without any state)

![image](https://github.com/apache/airflow/assets/122443108/076eb1e2-bbbb-489f-8ab5-35480c6820db)

As you can see in the image, all the task are 'found' but not loaded on the grid. Even all the previous task are finished succesfully.

Also if I check LOGS for previous NONE status task, logs are shown.

![image](https://github.com/apache/airflow/assets/122443108/5a06b922-0e2e-4cad-b546-46dd2bf8ecaa)

We are also seeing other behavior. Since we 'lose' the previous tasks, the DagRuns never finish and this results in the next DagRuns never being executed. That is, our Dummy task (Start) is not launched because since we do not have the previous tasks, we do not have their states.

--------------
Other example.

This is the GRID from a DAG.

![image](https://github.com/apache/airflow/assets/122443108/d0b5b98b-b91d-4dfd-900c-f0df54198478)

Also If I check for previous dags, this is what is shown.

![image](https://github.com/apache/airflow/assets/122443108/406232df-507f-4b8f-92f6-cbd4cba27143)

If I mark as success all 'start' task, suddenly GRID shows previous task, in this case with it's correct state.

![image](https://github.com/apache/airflow/assets/122443108/ed95251f-1d53-4fba-a18e-498c1f11068d)

### What you think should happen instead

All the task should be displayed correctly with their correct state

### How to reproduce

We don't know, we have days where all the DagRuns are with their correct tasks and status. Other cases where having the DAG correctly starts to go wrong and we end up losing the tasks. And other cases where nothing is affected.

### Operating System

"Debian GNU/Linux 11 (bullseye)" NAME="Debian GNU/Linux" 

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==8.0.0
apache-airflow-providers-celery==3.1.0
apache-airflow-providers-cncf-kubernetes==6.1.0
apache-airflow-providers-common-sql==1.4.0
apache-airflow-providers-docker==3.6.0
apache-airflow-providers-elasticsearch==4.4.0
apache-airflow-providers-ftp==3.3.1
apache-airflow-providers-google==10.0.0
apache-airflow-providers-grpc==3.1.0
apache-airflow-providers-hashicorp==3.3.1
apache-airflow-providers-http==4.3.0
apache-airflow-providers-imap==3.1.1
apache-airflow-providers-microsoft-azure==6.0.0
apache-airflow-providers-mysql==5.0.0
apache-airflow-providers-odbc==3.2.1
apache-airflow-providers-postgres==5.4.0
apache-airflow-providers-redis==3.1.0
apache-airflow-providers-sendgrid==3.1.0
apache-airflow-providers-sftp==4.2.4
apache-airflow-providers-slack==7.2.0
apache-airflow-providers-snowflake==4.0.5
apache-airflow-providers-sqlite==3.3.2
apache-airflow-providers-ssh==3.6.0

### Deployment

Official Apache Airflow Helm Chart

### Anything else

After upgrading from 2.5.1 to 2.6.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by PhoenixChamb** [comment]
We have lowered the airflow version to 2.5.3 and it has worked correctly again.

**Comment by bbovenzi** [comment]
Does this happen with all of your dags or just specific ones?

**Comment by PhoenixChamb** [comment]
@bbovenzi 

- Only in specific ones, not all the DAGS.
- We have not found any error both in the web server and in the programmer, which indicates a failure.
- Generally once we press the clear, past, downstream button. And finally we do not give. These tasks appear again, without status. In addition, usually the only tasks that we observe are those of the last DagRun.

**Comment by vDMG** [comment]
Hello, we have the exact same issue on `v2.6.3`. Happened like 2 days after upgrading from `2.6.2` to `2.6.3`.

![image](https://github.com/apache/airflow/assets/11329133/b3f6392d-c081-4bb6-a169-a20622895ab7)

I can observe some logs on PostgreSQL 
```
2023-07-26 12:57:54 UTC:X.X.X.X(X):user@db:[32269]:LOG: SSL error: decryption failed or bad record mac
2023-07-26 12:57:54 UTC:X.X.X.X(X):user@db:[32269]:LOG: could not receive data from client: Connection reset by peer
```

**Comment by vDMG** [comment]
Hi there,

We have been able to reproduce the incident and have determined that it is related to the use of a feature that allows users to delete multiple DAG runs at once with a selection. When this feature is used, it will delete the DAG run IDs for the selected DAGs, as well as the DAG run IDs for other DAGs and associated task instances.

![image](https://github.com/apache/airflow/assets/11329133/66379f59-de35-4d83-aa74-8258b2c4477e)

**Comment by potiuk** [comment]
From slack discussion https://apache-airflow.slack.com/archives/CCQ7EGB1P/p1691057005737649
Looks like duplicate of https://github.com/apache/airflow/pull/32684 and scheduled to be fixed in 2.7.0

Marking it as duplicate and closing for now - we can always re-open if it turns to be red-herring.
